### PR TITLE
svengato intermine-ws-python enhancements

### PR DESCRIPTION
**Pull request by svengato**
I am using the intermine python module to return templates from multiple InterMines. These changes are required to extract all the information I need.

### CODE REVIEW

File: intermine/query.py [hunk 1]
```diff
@@ -423,13 +424,20 @@ def from_xml(cls, xml, *args, **kwargs):
         doc = minidom.parse(f)
         f.close()
 
+        templates = doc.getElementsByTagName('template')
+        if len(templates) != 1:
+            raise QueryParseError("wrong number of templates in xml. "
+                + "Only one <template> element is allowed. Found %d" % len(templates))
```
Comment (yochannah) [inline intermine/query.py]: What query should I run to test this?

**Comment by yochannah** [comment]
p.s. - it is possible to fix the pep8 issues using the pep8 auto fix command in the readme. https://github.com/intermine/intermine-ws-python#before-making-prs

I also notice it's your first PR on GitHub - congrats! If you are thinking to make a few more (to InterMine or elsewhere you coud sign up for hacktoberfest and get a free tshirt: https://hacktoberfest.digitalocean.com/ )

File: intermine/query.py (hunk 1 above)
Comment (svengato) [reply intermine/query.py]: Lines 427 and 431-432 are for extracting obj.title from the template XML.
Lines 428-430 are simple error checking, exactly analogous to that for the query XML a few lines below.

**Comment by yochannah** [comment]
@svengato thanks for updating this to work nicelt with PEP8! It looks like there are a couple of test failures - the new error text you wrote is being output in our logs: https://travis-ci.org/intermine/intermine-ws-python/jobs/598365963#L3101 

Are you able to take a look?

**Comment by svengato** [comment]
I just noticed that Template is a subclass of Query, so it seems more logical to handle parsing the template title there. Let me experiment with that and get back to you -

**Comment by svengato** [comment]
It was a bit tricky as from_xml() is a Query class method, but this version should not generate the same error when parsing a non-template query.

**Pull request merged**